### PR TITLE
Pri0 and pri1 rework

### PR DESCRIPTION
**Pull request by jashook**
**Changes all outerloop jobs to explicitly use pri1 builds of the tests. It also rebrands PR Triggered pr triggered jobs to "innerloop" and will run pri0 tests.**

The change also includes netci.groovy cleanup. The list of Pr triggered jobs will change from:

Old PR Triggers (17):

```
CentOS7.1 x64 Debug Build and Test (debug_centos7.1_prtest)
CentOS7.1 x64 Release Priority 1 Build and Test (x64_release_centos7.1_pri1_flow_prtest)
OSX10.12 x64 Checked Build and Test (checked_osx10.12_flow_prtest)
Tizen armel Cross Debug Build (armel_cross_debug_tizen_prtest)
Tizen armel Cross Release Build (armel_cross_release_tizen_prtest)
Ubuntu arm64 Cross Debug Build (arm64_debug_small_page_size_prtest)
Ubuntu armlb Cross Release Build (armlb_cross_release_ubuntu_prtest)
Ubuntu x64 Checked Build and Test (checked_ubuntu_flow_prtest)
Ubuntu x64 Formatting (x64_ubuntu_formatting_prtest)
Ubuntu16.04 armlb Cross Debug Build (armlb_cross_debug_ubuntu16.04_prtest)
Windows_NT arm Cross Checked Build and Test (arm_cross_checked_windows_nt_prtest)
Windows_NT arm64 Cross Debug Build (arm64_cross_debug_windows_nt_prtest)
Windows_NT armlb Cross Checked Build and Test (armlb_cross_checked_windows_nt_prtest)
Windows_NT x64 Debug Build and Test (debug_windows_nt_prtest)
Windows_NT x64 Formatting (x64_windows_nt_formatting_prtest)
Windows_NT x64 Release Priority 1 Build and Test (x64_release_windows_nt_pri1_prtest)
Windows_NT x86 Checked Build and Test (x86_checked_windows_nt_prtest)
```

New PR Triggers (15):

```
CentOS7.1 x64 Checked Innerloop Build and Test (checked_centos7.1_flow_prtest)
CentOS7.1 x64 Debug Innerloop Build (debug_centos7.1_prtest)
OSX10.12 x64 Checked Innerloop Build and Test (checked_osx10.12_flow_prtest)
Ubuntu arm64 Debug Build (arm64_debug_small_page_size_prtest)
Ubuntu armlb Innerloop Cross Debug Build (armlb_cross_debug_ubuntu_prtest)
Ubuntu x64 Checked Innerloop Build and Test (checked_ubuntu_flow_prtest)
Ubuntu x64 Innerloop Formatting (x64_ubuntu_formatting_prtest)
Ubuntu16.04 armlb Innerloop Cross Debug Build (armlb_cross_debug_ubuntu16.04_prtest)
Windows_NT arm Cross Checked Innerloop Build and Test (arm_cross_checked_windows_nt_prtest)
Windows_NT arm64 Cross Checked Innerloop Build and Test (arm64_cross_checked_windows_nt_prtest)
Windows_NT arm64 Cross Debug Innerloop Build (arm64_cross_debug_windows_nt_prtest)
Windows_NT armlb Cross Checked Innerloop Build and Test (armlb_cross_checked_windows_nt_prtest)
Windows_NT x64 Checked Innerloop Build and Test (checked_windows_nt_prtest)
Windows_NT x64 Innerloop Formatting (x64_windows_nt_formatting_prtest)
Windows_NT x86 Checked Innerloop Build and Test (x86_checked_windows_nt_prtest)
```

PR Trigger Change summary:

```
-- Added Jobs --
CentOS7.1 x64 Checked Innerloop Build and Test (checked_centos7.1_flow_prtest)
Ubuntu armlb Cross Innerloop Debug Build (armlb_cross_debug_ubuntu_prtest)
Windows_NT arm64 Cross Checked Innerloop Build and Test (arm64_cross_checked_windows_nt_prtest)
Windows_NT x64 Checked Innerloop Build and Test (checked_windows_nt_prtest)

-- Removed Jobs --
CentOS7.1 x64 Release Priority 1 Build and Test (x64_release_centos7.1_pri1_flow_prtest)
Tizen armel Cross Debug Build (armel_cross_debug_tizen_prtest)
Tizen armel Cross Release Build (armel_cross_release_tizen_prtest)
Ubuntu armlb Cross Release Build (armlb_cross_release_ubuntu_prtest)
Windows_NT x64 Debug Build and Test (debug_windows_nt_prtest)
Windows_NT x64 Release Priority 1 Build and Test (x64_release_windows_nt_pri1_prtest)

-- Changed Jobs --
CentOS7.1 x64 Debug Build and Test -> CentOS7.1 x64 Debug Innerloop Build (debug_centos7.1_prtest)
OSX10.12 x64 Checked Build and Test -> OSX10.12 x64 Checked Innerloop Build and Test (checked_osx10.12_flow_prtest)
Ubuntu arm64 Cross Debug Build -> Ubuntu arm64 Cross Debug Innerloop Build
Ubuntu x64 Checked Build and Test -> Ubuntu x64 Checked Innerloop Build and Test (checked_ubuntu_flow_prtest)
Ubuntu x64 Formatting -> Ubuntu x64 Innerloop Formatting (x64_ubuntu_formatting_prtest)
Ubuntu16.04 armlb Cross Debug Build -> Ubuntu16.04 armlb Cross Debug Innerloop Build (armlb_cross_debug_ubuntu16.04_prtest)
Windows_NT arm Cross Checked Build and Test -> Windows_NT arm Cross Checked Innerloop Build and Test (arm_cross_checked_windows_nt_prtest)
Windows_NT arm64 Cross Debug Build -> Windows_NT arm64 Cross Debug Innerloop Build (arm64_cross_debug_windows_nt_prtest)
Windows_NT armlb Cross Checked Build and Test -> Windows_NT armlb Cross Checked Innerloop Build and Test (armlb_cross_checked_windows_nt_prtest)
Windows_NT x64 Formatting -> Windows_NT x64 Innerloop Formatting (x64_windows_nt_formatting_prtest)
Windows_NT x86 Checked Build and Test -> Windows_NT x86 Checked Innerloop Build and Test (x86_checked_windows_nt_prtest)
```

### CODE REVIEW

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -1576,26 +1511,18 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
...
-                    // For Pri 1 tests, we must shorten the output test binary path names.
-                    // if __TestIntermediateDir is already set, build-test.cmd will
-                    // output test binaries to that directory. If it is not set, the
-                    // binaries are sent to a default directory whose name is about
-                    // 35 characters long.
-
-                    else if (scenarioNeedsPri1Build(scenario)) {
-                        buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${arch} ${buildOpts} -priority=1"
+                        buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${arch} ${buildOpts} -priority=${priority}"
                     }
```
Comment (BruceForstall) [inline netci.groovy]: you've already added `-priority` to `buildOpts` above; looks like it's getting passed twice. (same below)

File: netci.groovy [hunk 2]
```diff
@@ -1576,26 +1511,18 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
...
                         buildOpts += enableCorefxTesting ? ' skiptests' : ''
-                        buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${arch} ${buildOpts}"
-                    }
-
-                    // For Pri 1 tests, we must shorten the output test binary path names.
-                    // if __TestIntermediateDir is already set, build-test.cmd will
-                    // output test binaries to that directory. If it is not set, the
-                    // binaries are sent to a default directory whose name is about
-                    // 35 characters long.
-
```
Comment (BruceForstall) [inline netci.groovy]: You should probably keep this comment, or else the code that sets `__TestIntermediateDir` will be confusing.

File: netci.groovy [hunk 3]
```diff
@@ -1576,26 +1511,18 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
...
-                    else if (scenarioNeedsPri1Build(scenario)) {
-                        buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${arch} ${buildOpts} -priority=1"
+                        buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${arch} ${buildOpts} -priority=${priority}"
                     }
                     else if (isLongGc(scenario)) {
                         buildCommands += "build.cmd ${lowerConfiguration} ${arch} ${buildOpts} skiptests"
-                        buildCommands += "set __TestIntermediateDir=int&&build-test.cmd ${lowerConfiguration} ${arch}"
+                        buildCommands += "set __TestIntermediateDir=int&&build-test.cmd ${lowerConfiguration} ${arch} -priority=${priority}"
                     }
```
Comment (BruceForstall) [inline netci.groovy]: (pre-existing issue:) There must be a reasons the build and build-test invocations are separate, but it's not obvious why to me.

**Comment by BruceForstall** [comment]
I don't have many comments or context, but it basically LGTM

File: netci.groovy (hunk 2 above)
Comment (jashook) [reply netci.groovy]: Addressed

**Comment by jashook** [comment]
@swgillespie this change touches some of the long gc jobs. As the change is, it will build priority 1 tests for every outlerloop job, including those jobs. Do you have any objections?

**Comment by jashook** [comment]
Also /cc @wtgodbe

File: netci.groovy [hunk 4]
```diff
@@ -1563,6 +1491,13 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
                         buildCommands += "tests\\scripts\\build_illink.cmd clone ${arch}"
                     }
 
+                    def priority = '1'
+                    if (scenario == 'default') {
+                        priority = '0'
+                    }
+
+                    buildOpts += ' -priority=${priority}'
```
Comment (mmitche) [inline netci.groovy]: Double quoted string here.

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
/cc @dotnet/jit-contrib.

This will **change all outerloop jobs to explicitly use pri1 builds of the tests. It also rebrands PR Triggered pr triggered jobs to "innerloop" and will run pri0 tests.**

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by jashook** [comment]
@dotnet-bot test ci please

**Comment by hqueue** [comment]
@jashook Can we re-enable Tizen armel CIs  ? Becuase `armel` uses different implementation of JIT compiler, we need it to maintain `armel`(softfp ABI) port.

```
Tizen armel Cross Debug Build (armel_cross_debug_tizen_prtest)
Tizen armel Cross Release Build (armel_cross_release_tizen_prtest)
```

**Pull request merged**